request_test.go

```diff
@@ -247,3 +249,108 @@ func TestOpendirHandleReuse(t *testing.T) {
 	rpkt = request.call(handlers, pkt, nil, 0)
 	assert.IsType(t, &sshFxpNamePacket{}, rpkt)
 }
+
+func Test_toLocalPath(t *testing.T) {
+	type args struct {
+		workDir string
+		p       string
+	}
+	tests := []struct {
+		name string
+		goos string
+		args args
+		want string
+	}{
+		{
+			name: "empty path",
+			goos: "linux",
+			args: args{p: ""},
+			want: "",
+		},
+		{
+			name: "relative path",
+			goos: "linux",
+			args: args{p: "file"},
+			want: "file",
+		},
+		{
+			name: "absolute path",
+			goos: "linux",
+			args: args{p: "/file"},
+			want: "/file",
+		},
+		{
+			name: "empty path",
+			goos: "linux",
+			args: args{workDir: "/home/user", p: ""},
+			want: "/home/user",
+		},
+		{
+			name: "relative path",
+			goos: "linux",
+			args: args{workDir: "/home/user", p: "file"},
+			want: "/home/user/file",
+		},
+		{
+			name: "relative path with .",
+			goos: "linux",
+			args: args{workDir: "/home/user", p: "."},
+			want: "/home/user",
+		},
+		{
+			name: "absolute path",
+			goos: "linux",
+			args: args{workDir: "/home/user", p: "/file"},
+			want: "/file",
+		},
+		{
+			name: "empty path",
+			goos: "windows",
+			args: args{workDir: "C:\\Users\\User", p: ""},
+			want: "C:\\Users\\User",
+		},
+		{
+			name: "relative path",
+			goos: "windows",
+			args: args{workDir: "C:\\Users\\User", p: "file"},
+			want: "C:\\Users\\User\\file",
+		},
+		{
+			name: "relative path with .",
+			goos: "windows",
+			args: args{workDir: "C:\\Users\\User", p: "."},
+			want: "C:\\Users\\User",
+		},
+		{
+			name: "absolute path",
+			goos: "windows",
+			args: args{workDir: "C:\\Users\\User", p: "C:\\file"},
+			want: "C:\\file",
+		},
+		{
+			name: "relative unix-like path",
+			goos: "windows",
+			args: args{workDir: "C:\\Users\\User", p: "./file"},
+			want: "C:\\Users\\User\\file",
+		},
+		{
+			name: "absolute unix-like path",
+			goos: "windows",
+			args: args{workDir: "C:\\Users\\User", p: "/C:/file"},
+			want: "C:\\file",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf("%s %s %s", tt.goos, tt.args.workDir, tt.name), func(t *testing.T) {
+			if runtime.GOOS != tt.goos {
+				t.Skipf("Skipping test for %s on %s", tt.goos, runtime.GOOS)
+			}
```

Comment:
Prefer to not skipping tests like this. There are already build tags available, which will exclude the tests or not depending on the filename of the test.

Reply:
I can move this to separate files, but then the test body would have to be duplicated across those files. Skipping to match the platform seems like a lesser bad to me?

Reply:
> New files called request_windows_test.go and request_unix_test.go?

Yes please.

> I looked at the codebase and there's already a lot of t.Skip present, and barely any test files utilizing build flags.

Sadly, there’s a lot of inherited code that I would write differently now.